### PR TITLE
refactor to use react-navigation v5 with hooks

### PR DESCRIPTION
**Pull request by amandeepmittal**
Hi @byCedric,

This PR solves #10 & #3 with some additions and a lot of refactoring to include:
- `react-navigation` v5 usage
- Only use user's context through Context API
- space to add custom or more providers
- custom statusbar hook for each screen using `useFocusEffect()` from react-navigation
- a welcome screen
- reusable form components using Formik
- usual form validation using Yup
- Firebase auth error handling
- remove `react-native-elements` to create UI instead add custom UI components

Let me know what you think about this. I will update the document in another PR.

### CODE REVIEW

File: navigation/Routes.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+      try {
+        await (authUser ? setUser(authUser) : setUser(null));
+        setIsLoading(false);
+      } catch (error) {
+        console.log(error);
+      }
+    });
+
+    // unsubscribe listener on unmount
+    return () => authListener;
```
Comment (byCedric) [inline navigation/Routes.js:26]: What is `authListener` doing here, if it's a method to unregister the listener you have to invoke it, right?

Comment (amandeepmittal) [reply navigation/Routes.js:26]: Yes, here the listener is getting unsubscribed. Is there any other way to unsubscribe it? I am using the same strategy with React Native vanilla apps using `react-native-firebase` package.

Comment (byCedric) [reply navigation/Routes.js:26]: It's a hard-to-spot issue, but I ran into this one a couple of times when creating hooks for Expo. I think what you actually wanted to do here is `return authListener;`. Once the hook unmounts, the unsubscriber is called ([see this SO](https://stackoverflow.com/questions/37370224/firebase-stop-listening-onauthstatechanged)).

You can also write it as `() => authListener()` ofc., but that's eventually the same thing. You only add an extra method 😄 Maybe it's also good to rename the variable to something like `unsubscribeAuth` or `removeAuthListener`. Should make it a bit more understandable as well.

Comment (amandeepmittal) [reply navigation/Routes.js:26]: Thanks for sharing this and taking a look at this, @byCedric. I will update it likewise both based on the correct form of returning an unsubscribe function returning from the `onAuthStateChanged`  as well use better naming convention.

**Pull request merged**